src/foam/nanos/auth/UserAndGroupAuthService.java

```diff
@@ -155,6 +155,10 @@ public void start() {
     challengeMap  = new LRULinkedHashMap<Long, Challenge>(20000);
   }
 
+  public User getCurrentUser() {
```

Comment:
First argument of getCurrentUser() will need to be X, so that it's set by the SessionServerBox so that you actually know the current User.

Reply:
Thanks, added